src/App.tsx

```diff
@@ -1,24 +1,209 @@
+  const [filterType, setFilterType] = useState<FilterParam>(FilterParam.All);
+  const [errorType, setErrorType] = useState(ErrorType.None);
+  const [title, setTitle] = useState('');
+  const [tempTodo, setTempTodo] = useState<null | Todo>(null);
+  const [loadingTodo, setLoadingTodo] = useState([0]);
+  const [isSending, setIsSending] = useState(false);
+  const [
+    countActiveTodo,
+    setCountActiveTodo,
+  ] = useState(counterOfActiveTodos(todos));
+  const [isToggleAll, setIsToggleAll] = useState(false);
+  const [isOnRender, setIsOnRender] = useState(false);
+
+  const handleTextChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+  };
+
+  const fetchTodos = async () => {
+    try {
+      const todosFromServer = await getTodos(USER_ID);
+
+      setTodos(todosFromServer);
+    } catch (error) {
+      setHasError(true);
+      setErrorType(ErrorType.Load);
+    }
+  };
+
+  const addNewTodo = useCallback((newTodo: Todo): void => {
+    setTodos((oldTodos) => [...oldTodos, newTodo]);
+  },
+  []);
+
+  const showError = useCallback((error: ErrorType) => {
+    setErrorType(error);
+    setHasError(true);
+  }, []);
+
+  const hideError = useCallback(() => {
+    setHasError(false);
+  }, []);
+
+  const removeTodo = async (todoId: number) => {
+    try {
+      setLoadingTodo(prevTodo => [...prevTodo, todoId]);
+
+      await deleteTodo(todoId);
+      setTodos(todoForDelete => (
+        todoForDelete.filter(todo => todo.id !== todoId)
+      ));
+    } catch {
+      setHasError(true);
+      showError(ErrorType.Delete);
+    } finally {
+      setLoadingTodo([0]);
+    }
+  };
+
+  const addNewTodoInList = async (newTodo: Todo) => {
+    if (!isSending) {
+      try {
+        const downloadNewTodo = await addTodo(USER_ID, newTodo);
+
+        addNewTodo(downloadNewTodo);
+      } catch {
+        showError(ErrorType.Add);
+      } finally {
+        setTitle('');
+        setTempTodo(null);
+        setIsSending(false);
+      }
+    }
+  };
+
+  const handleFormSubmit = (event: React.FocusEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!title.trim()) {
+      showError(ErrorType.EmptyTitle);
+      setTitle('');
+
+      return;
+    }
+
+    const newTodo = {
+      userId: USER_ID,
+      title,
+      completed: false,
+      id: 0,
+    };
+
+    setTempTodo({ ...newTodo, id: 0 });
+    setIsSending(true);
+
+    addNewTodoInList(newTodo);
+  };
+
+  const updateTodo = async (
+    todoId: number,
+    completed?: boolean,
+    newTitle?: string,
+  ) => {
+    setLoadingTodo([todoId]);
+    try {
+      await patchTodo(todoId, completed, newTitle);
+      fetchTodos();
+    } catch (error) {
+      showError(ErrorType.Update);
+      setHasError(true);
+      setIsOnRender(true);
+    } finally {
+      setLoadingTodo([0]);
+      setIsToggleAll(false);
+      setIsOnRender(false);
+    }
+  };
+
+  const toggleCompletedAllTodo = () => {
+    todos.map(todo => {
```

Comment:
Use forEach instead of map here, as map should be used when you transform each element of an array and return a new array. You're not using the returned value of map in this case.